reana_commons/config.py

```diff
@@ -196,3 +196,16 @@
 
 K8S_REANA_SERVICE_ACCOUNT_NAME = os.getenv('K8S_REANA_SERVICE_ACCOUNT_NAME')
 """REANA service account in the deployed Kubernetes cluster."""
+
+
+# REANA_COMPONENTS_PREFIX = os.getenv('REANA_COMPONENTS_PREFIX', 'cern')
```

Comment:
Let us take two possible prefixes to illustrate the two naming schemes, `cms` and `my-reana`:
- Scheme: `<helm-release-prefix>-reana-<component-name>` which leads to -> `cms-reana-server` or `my-reana-reana-server`
- Scheme `<helm-release-prefix>-<component-name>` which leads to -> `cms-server` or `my-reana-server`
- Others...

Reply:
This naming logic is unfortunately spread in different places:
- Helm charts
    - Component names
    - Shared volumes
    - Configuration files (krb5)
- REANA code
  - Secrets (k8s secrets)
  - Workflows/interactive sessions (k8s jobs, deployments, ingresses etc..)
  - Kubernetes jobs (k8s jobs)

To be improved.

Reply:
(1) Regarding naming schemes, I think the second option looks good:

- `<helm_release_prefix>-<component-name>`

where `helm_release_prefix` would default to `reana` if nothing is specified by the user.

(2) We could also take advantage and shorten the workflow runtime pod names:

- `reana-interactive-jupyter-<sha1>` -> `reana-int-jup-...`

- `reana-batch-cwl-<sha1>` -> `reana-bat-cwl-...`
- `reana-batch-serial-<sha1>` -> `reana-bat-ser-...`
- `reana-batch-yadage-<sha1>` -> `reana-bat-yad-...`

following `reana-XXX-YYY-...` naming schema? Three characters should be enough for foreseeable future.  We mused about shortening pod names long time ago,so if it helps with long helm prefixes, it could perhaps be considered afterwards after the current PR set works?

(Well, it is especially `interactive-jupyter-...` that is long, for workflows it helps less, but if we add more engines, it seems still advantageous to shorten it.)

Reply:
Regarding (1) I agree, I will go for this.

Regarding (2): what about using labels to put this information, and just have a generic workflow and session (just suggestions, feel free to change):
- `reana-session-<uuid>` where we would have labels such as:
  - `session-type=[jupyter|terminal|...]`
  - `...=...`
- `reana-workflow-<uuid>` weher we would have labels such as:
  - `workflow-engine=[cwl|yadage|serial...]`
  - `...=...`

Side note: Code harmony wise is not that bad
```python
...
    def start_interactive_session(self):
        """Start an interactive workflow run."""
        raise NotImplementedError('')

    def stop_batch_workflow_run(self):
        """Stop a batch workflow run."""
        raise NotImplementedError('')
...
```

Practically speaking, instead of `grep`, we would use [label filtering](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#api) locally and in production, we would rely on [Prometheus and Grafana](https://clouddocs.web.cern.ch/containers/tutorials/prometheus.html).

Reply:
Fully agree with using labels :+1: We could use labels even more widely, e.g. `name=roofit,run=10.2,user=johndoe,engine=cwl`, so that we could easily filter pods e.g. to find all pods of some user, e.g. to find all pods based on some human-readable workflow name.

If the labels would be also applied to jobs (?) and if we could use them easily in dashboard (?) then this could also provide some poor-man's accounting overview for a start...

Reply:
Looks like this:
```console
$ kubectl get pods
my-reana-workflow-controller-54c44bcb59-fwsqg       2/2     Running             0          3s                      
my-reana-traefik-7c887cbcf6-4x8jm                   0/1     Running             0          6s                      
my-reana-wdb-cb99848dc-jds88                        1/1     Running             0          6s                                                                                                                                         
my-reana-message-broker-7b7744c97b-ksq8q            1/1     Running             0          6s                      
my-reana-db-cc446cbf8-fwqlq                         1/1     Running             0          6s                      
my-reana-cache-7cb44b76c-8hnh2                      1/1     Running             0          6s                      
my-reana-server-79996fbb6b-vmq7x                    2/2     Running             0          6s                      
my-reana-traefik-7c887cbcf6-4x8jm                   1/1     Running             0          14s                                                                                                                                        
my-reana-run-batch-c7a74054-3fb4-4f6a-9c92-7c9b6c78a189-xht9w   0/2     Pending             0          0s          
my-reana-run-batch-c7a74054-3fb4-4f6a-9c92-7c9b6c78a189-xht9w   0/2     Pending             0          0s          
my-reana-run-batch-c7a74054-3fb4-4f6a-9c92-7c9b6c78a189-xht9w   0/2     ContainerCreating   0          0s          
my-reana-run-batch-c7a74054-3fb4-4f6a-9c92-7c9b6c78a189-xht9w   2/2     Running             0          1s                                                                                                                             
my-reana-run-job-b7afc273-b477-48e8-941b-a7b5f6bf608b-8vxtr     0/1     Pending             0          0s          
my-reana-run-job-b7afc273-b477-48e8-941b-a7b5f6bf608b-8vxtr     0/1     Pending             0          0s          
my-reana-run-job-b7afc273-b477-48e8-941b-a7b5f6bf608b-8vxtr     0/1     ContainerCreating   0          0s          
my-reana-run-job-b7afc273-b477-48e8-941b-a7b5f6bf608b-8vxtr     0/1     Completed           0          2s          
my-reana-run-job-b7afc273-b477-48e8-941b-a7b5f6bf608b-8vxtr     0/1     Terminating         0          2s          
my-reana-run-job-b7afc273-b477-48e8-941b-a7b5f6bf608b-8vxtr     0/1     Terminating         0          2s          
my-reana-run-batch-c7a74054-3fb4-4f6a-9c92-7c9b6c78a189-xht9w   2/2     Terminating         0          12s         
my-reana-run-batch-e227c458-801f-4ab7-b326-186b77e03c7f-nrl4x   0/2     Pending             0          0s                                                                                                                             
my-reana-run-batch-e227c458-801f-4ab7-b326-186b77e03c7f-nrl4x   0/2     Pending             0          0s          
my-reana-run-batch-e227c458-801f-4ab7-b326-186b77e03c7f-nrl4x   0/2     ContainerCreating   0          1s          
my-reana-run-batch-e227c458-801f-4ab7-b326-186b77e03c7f-nrl4x   2/2     Running             0          2s          
my-reana-run-job-8f785bbc-bdfb-4b1b-940c-d26e1b258e8f-r8m9n     0/1     Pending             0          0s     
my-reana-run-session-e227c458-801f-4ab7-b326-186b77e03c7f-7gtjt   0/1     Pending             0          0s        
my-reana-run-session-e227c458-801f-4ab7-b326-186b77e03c7f-7gtjt   0/1     Pending             0          1s        
my-reana-run-session-e227c458-801f-4ab7-b326-186b77e03c7f-7gtjt   0/1     ContainerCreating   0          1s 
```